docs/IE_PLUGIN_DG/InferRequest.md

```diff
@@ -1,76 +1,65 @@
+> **NOTE**: Use inputs/outputs information from the compiled model to understand shape and element type of tensors, which you can set with ov::InferRequest::set_tensor and get with ov::InferRequest::get_tensor. A plugin uses these hints to determine its internal layouts and element types for input and output blobs if needed. 
 
-### `~InferRequest` Destructor
+### ~InferRequest Destructor
 
-Decrements a number of created inference requests: 
+Destructor can contain plugin specific logic to finish and destroy infer request.
 
 @snippet src/sync_infer_request.cpp infer_request:dtor
 
-### `InferImpl()`
+### infer()
 
-**Implementation details:** Base IInferRequestInternal class implements the public InferenceEngine::IInferRequestInternal::Infer method as following:
-- Checks blobs set by users
-- Calls the `InferImpl` method defined in a derived class to call actual pipeline stages synchronously
+The method calls actual pipeline stages synchronously. Inside the method plugin should check input/output tensors, move external tensors to backend and run the inference.
 
 @snippet src/sync_infer_request.cpp infer_request:infer_impl
 
-#### 1. `inferPreprocess`
+#### 1. infer_preprocess()
 
-Below is the code of the `inferPreprocess` method to demonstrate Inference Engine common preprocessing step handling:
+Below is the code of the `infer_preprocess()` method. The method checks user input/output tensors and demonstrates conversion from user tensor to backend specific representation:
 
 @snippet src/sync_infer_request.cpp infer_request:infer_preprocess
 
-**Details:**
-* `InferImpl` must call the InferenceEngine::IInferRequestInternal::execDataPreprocessing function, which executes common Inference Engine preprocessing step (for example, applies resize or color conversion operations) if it is set by the user. The output dimensions, layout and precision matches the input information set via InferenceEngine::CNNNetwork::getInputsInfo.
-* If `inputBlob` passed by user differs in terms of precisions from precision expected by plugin, `blobCopy` is performed which does actual precision conversion.
+#### 2. start_pipeline()
 
-#### 2. `startPipeline`
-
-Executes a pipeline synchronously using `_executable` object:
+Executes a pipeline synchronously using `m_executable` object:
 
 @snippet src/sync_infer_request.cpp infer_request:start_pipeline
 
-#### 3. `inferPostprocess`
+#### 3. infer_postprocess()
 
-Converts output blobs if precisions of backend output blobs and blobs passed by user are different:
+Converts backend specific tensors to tensors passed by user:
 
 @snippet src/sync_infer_request.cpp infer_request:infer_postprocess
 
```

Comment:
looks like description for get_performance_counts and set_tensors_impl is missed in this documentation